### ISSUE TITLE
Problem on using WandB with YOLOv8

### ISSUE THREAD

**Issue by HazmanNaim** [ultralytics/ultralytics#8614]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello. I am training a YOLOv8 model on Kaggle. I am following a sample notebook that implementing WandB with YOLOv8. Here is the example notebook I am following: [example notebook](https://www.kaggle.com/code/mersico/medical-instance-segmentation-with-yolov8/notebook#YOLOv8-Architecture:-A-Deep-Dive).

However, when I start to train the model, I got the error which is related to `wandb.sdk.interface.interface_sock`. Here are the full error including the code:

```
# Load a model
model = YOLO("yolov8s.pt")  # load a pretrained model (recommended for new training)

# Train the model
results = model.train(data="processed_yolo/dataset.yaml",
                      project = "storm_ml_training",
                      patience =15,
                      epochs=100, 
                      imgsz=(512, 512))
```

Error:

```
Ultralytics YOLOv8.1.20 🚀 Python-3.10.13 torch-2.1.2 CUDA:0 (Tesla P100-PCIE-16GB, 16276MiB)
engine/trainer: task=detect, mode=train, model=yolov8s.pt, data=processed_yolo/dataset.yaml, epochs=100, time=None, patience=15, batch=16, imgsz=(512, 512), save=True, save_period=-1, cache=False, device=None, workers=8, project=storm_ml_training, name=train2, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, auto_augment=randaugment, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=storm_ml_training/train2
Overriding model.yaml nc=80 with nc=4

                   from  n    params  module                                       arguments                     
  0                  -1  1       928  ultralytics.nn.modules.conv.Conv             [3, 32, 3, 2]                 
  1                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  2                  -1  1     29056  ultralytics.nn.modules.block.C2f             [64, 64, 1, True]             
  3                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  4                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  5                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  6                  -1  2    788480  ultralytics.nn.modules.block.C2f             [256, 256, 2, True]           
  7                  -1  1   1180672  ultralytics.nn.modules.conv.Conv             [256, 512, 3, 2]              
  8                  -1  1   1838080  ultralytics.nn.modules.block.C2f             [512, 512, 1, True]           
  9                  -1  1    656896  ultralytics.nn.modules.block.SPPF            [512, 512, 5]                 
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  1    591360  ultralytics.nn.modules.block.C2f             [768, 256, 1]                 
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 
 16                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
 17            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 
 19                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
 20             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  1   1969152  ultralytics.nn.modules.block.C2f             [768, 512, 1]                 
 22        [15, 18, 21]  1   2117596  ultralytics.nn.modules.head.Detect           [4, [128, 256, 512]]          
Model summary: 225 layers, 11137148 parameters, 11137132 gradients, 28.7 GFLOPs

Transferred 349/355 items from pretrained weights
TensorBoard: Start with 'tensorboard --logdir storm_ml_training/train2', view at http://localhost:6006/
Changes to your `wandb` environment variables will be ignored because your `wandb` session has already started. For more information on how to modify your settings with `wandb.init()` arguments, please refer to [the W&B docs](https://wandb.me/wandb-init).
Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/wandb/sdk/wandb_init.py", line 1176, in init
  File "/opt/conda/lib/python3.10/site-packages/wandb/sdk/wandb_init.py", line 627, in init
  File "/opt/conda/lib/python3.10/site-packages/wandb/sdk/backend/backend.py", line 162, in ensure_launched
  File "/opt/conda/lib/python3.10/site-packages/wandb/sdk/backend/backend.py", line 150, in _ensure_launched_manager
ModuleNotFoundError: No module named 'wandb.sdk.interface.interface_sock'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/lib/python3.10/site-packages/wandb/sdk/wandb_init.py", line 1184, in init
  File "/opt/conda/lib/python3.10/site-packages/wandb/sdk/wandb_init.py", line 854, in getcaller
  File "/root/.local/lib/python3.10/site-packages/log.py", line 45, in findCaller
    traceback.print_stack(f, file=sio)
NameError: name 'traceback' is not defined
---------------------------------------------------------------------------
ModuleNotFoundError                       Traceback (most recent call last)
File /opt/conda/lib/python3.10/site-packages/wandb/sdk/wandb_init.py:1176, in init(job_type, dir, config, project, entity, reinit, tags, group, name, notes, magic, config_exclude_keys, config_include_keys, anonymous, mode, allow_val_change, resume, force, tensorboard, sync_tensorboard, monitor_gym, save_code, id, settings)

File /opt/conda/lib/python3.10/site-packages/wandb/sdk/wandb_init.py:627, in init(self)

File /opt/conda/lib/python3.10/site-packages/wandb/sdk/backend/backend.py:162, in ensure_launched(self)

File /opt/conda/lib/python3.10/site-packages/wandb/sdk/backend/backend.py:150, in _ensure_launched_manager(self)

ModuleNotFoundError: No module named 'wandb.sdk.interface.interface_sock'

During handling of the above exception, another exception occurred:

NameError                                 Traceback (most recent call last)
File /opt/conda/lib/python3.10/site-packages/wandb/sdk/wandb_init.py:1184, in init(job_type, dir, config, project, entity, reinit, tags, group, name, notes, magic, config_exclude_keys, config_include_keys, anonymous, mode, allow_val_change, resume, force, tensorboard, sync_tensorboard, monitor_gym, save_code, id, settings)

File /opt/conda/lib/python3.10/site-packages/wandb/sdk/wandb_init.py:854, in getcaller()

File ~/.local/lib/python3.10/site-packages/log.py:45, in _Logger.findCaller(self, stack_info, stacklevel)
     44 sio.write('Stack (most recent call last):\n')
---> 45 traceback.print_stack(f, file=sio)
     46 sinfo = sio.getvalue()

NameError: name 'traceback' is not defined

The above exception was the direct cause of the following exception:

Error                                     Traceback (most recent call last)
Cell In[25], line 6
      2 model = YOLO("yolov8s.pt")  # load a pretrained model (recommended for new training)
      3 
      4 
      5 # Train the model
----> 6 results = model.train(data="processed_yolo/dataset.yaml",
      7                       project = "storm_ml_training",
      8                       patience =15,
      9                       epochs=100, 
     10                       imgsz=(512, 512))

File /opt/conda/lib/python3.10/site-packages/ultralytics/engine/model.py:644, in Model.train(self, trainer, **kwargs)
    641             pass
    643 self.trainer.hub_session = self.session  # attach optional HUB session
--> 644 self.trainer.train()
    645 # Update model and cfg after training
    646 if RANK in (-1, 0):

File /opt/conda/lib/python3.10/site-packages/ultralytics/engine/trainer.py:208, in BaseTrainer.train(self)
    205         ddp_cleanup(self, str(file))
    207 else:
--> 208     self._do_train(world_size)

File /opt/conda/lib/python3.10/site-packages/ultralytics/engine/trainer.py:322, in BaseTrainer._do_train(self, world_size)
    320 if world_size > 1:
    321     self._setup_ddp(world_size)
--> 322 self._setup_train(world_size)
    324 nb = len(self.train_loader)  # number of batches
    325 nw = max(round(self.args.warmup_epochs * nb), 100) if self.args.warmup_epochs > 0 else -1  # warmup iterations

File /opt/conda/lib/python3.10/site-packages/ultralytics/engine/trainer.py:235, in BaseTrainer._setup_train(self, world_size)
    232 """Builds dataloaders and optimizer on correct rank process."""
    234 # Model
--> 235 self.run_callbacks("on_pretrain_routine_start")
    236 ckpt = self.setup_model()
    237 self.model = self.model.to(self.device)

File /opt/conda/lib/python3.10/site-packages/ultralytics/engine/trainer.py:171, in BaseTrainer.run_callbacks(self, event)
    169 """Run all existing callbacks associated with a particular event."""
    170 for callback in self.callbacks.get(event, []):
--> 171     callback(self)

File /opt/conda/lib/python3.10/site-packages/ultralytics/utils/callbacks/wb.py:112, in on_pretrain_routine_start(trainer)
    110 def on_pretrain_routine_start(trainer):
    111     """Initiate and start project if module is present."""
--> 112     wb.run or wb.init(project=trainer.args.project or "YOLOv8", name=trainer.args.name, config=vars(trainer.args))

File /opt/conda/lib/python3.10/site-packages/wandb/sdk/wandb_init.py:1214, in init(job_type, dir, config, project, entity, reinit, tags, group, name, notes, magic, config_exclude_keys, config_include_keys, anonymous, mode, allow_val_change, resume, force, tensorboard, sync_tensorboard, monitor_gym, save_code, id, settings)

Error: An unexpected error occurred
```

**Comment by glenn-jocher** [comment]
@HazmanNaim hey there! 👋 It looks like you're encountering an issue with integrating WandB in your YOLOv8 training on Kaggle. The error message suggests there might be a problem with the WandB SDK installation or its initialization in your environment.

First, ensure that WandB is correctly installed and updated in your Kaggle notebook. You can do this by running:

```python
!pip install wandb --upgrade
```

Then, before starting your training, explicitly initialize WandB using:

```python
import wandb
wandb.login()
```

This should prompt you to log in to WandB and provide an API key if you haven't already done so in your environment.

After logging in, you can proceed with your training code as you have it. This explicit login step ensures that WandB is correctly initialized before starting the training process.

If the issue persists, it might be helpful to check the WandB documentation or reach out to their support for more specific guidance related to their SDK's integration.

Hope this helps! Let us know if you have any more questions. 😊

**Comment by HazmanNaim** [comment]
Hello, it's finally working. I just realized that I pip-installed the latest version of Ultralytics instead of the specific version 8.0.186 as outlined in your documentation.

```
!pip install ultralytics
```

I changed it to:

```
!pip install --upgrade ultralytics==8.0.186
```

I'm not really an expert, but if possible, could you explain why when we use a specific version it works, but when we use other versions, it may not work?

**Comment by glenn-jocher** [comment]
@HazmanNaim hey there! 🎉 I'm glad to hear it's working for you now. 

The reason specifying a specific version like `ultralytics==8.0.186` works better in some cases is due to compatibility. Each version of Ultralytics YOLOv8 is tested to work with certain dependencies and features. When you use the exact version mentioned in the documentation, you're aligning with the tested environment, ensuring everything runs smoothly. 

Using the latest version (`ultralytics`) without specifying might introduce changes or updates that haven't been tested with your current setup, leading to potential issues. It's a bit like making sure all the puzzle pieces fit perfectly by using the ones we know match. 🧩

Hope that clears things up! If you have any more questions, feel free to ask. Happy coding! 😊

**Comment by HazmanNaim** [comment]
Thanks for the explanation! That makes a lot of sense. I understand now why specifying a specific version is important for ensuring compatibility.

I appreciate your proactive approach and providing such valuable insights !✨

**Comment by glenn-jocher** [comment]
@HazmanNaim you're welcome! I'm really glad to hear that the explanation was helpful. Ensuring compatibility is key to a smooth experience, and I'm here to help make that happen. If you have any more questions or need further assistance, feel free to reach out. Happy coding! 🚀
